src/base/module/record/Record.js

```diff
@@ -80,7 +80,9 @@ class Record {
         return allRecords;
       })
       .catch((err)=>{
-        kintoneRC.deleteCursor({id: myCursor.id});
+        if (myCursor) {
+          kintoneRC.deleteCursor({id: myCursor.id});
```

Comment:
@youtr-s 
"kintoneRC.deleteCursor({id: myCursor.id})" returns a Promise also and can be faulty too, but there is no error handling.
It is similar to line 78 above, so I think we should also add error handling (catch) for them.

Reply:
@trinhhunganh 
I see.
I think modify codes on line 78, 84 as following.
```
kintoneRC.deleteCursor({id: myCursor.id})
  .catch(deleteErr => {
    throw deleteErr;
})
```

Reply:
@trinhhunganh 
I think on line 84 we should handle error but on line 78 we should not handle, 
because developer may confuse if getRecords response return with delete cursor error.

And please tell me why on line 76 this process was implemented,
I think this is unnecessary and this process cause unexpected error when records deleted after create cursor.

Reply:
@trinhhunganh 
>I think on line 84 we should handle error but on line 78 we should not handle,
>because developer may confuse if getRecords response return with delete cursor error.
I also considered about error handling of the process on line 84, I think developer may confuse if they catch originate error and deleteCursor error as same time.
So sdk don't need throw deleteCursor. 
What do you think about it?

Reply:
> And please tell me why on line 76 this process was implemented

This is implemented in case some records are delete after the cursor created. Then this function will return the fetched records and also delete the cursor.

Currently when using getAllRecordsByCursor, developers can't control deleting cursor. If there's something wrong deleting the cursor and cursor limit reached, developers will have to wait for 5m before they can create another cursor. However, I think it will not be a big problem so we can skip it.